src/decompressor.rs

```diff
@@ -0,0 +1,117 @@
+use std::ffi::OsStr;
+use std::io::Read;
+use std::path::Path;
+use std::collections::HashMap;
+use std::process::Command;
+use std::process::Stdio;
+use std::process::ChildStdout;
+use globset::{Glob, GlobSet, GlobSetBuilder};
+
+/// A decompression command, contains the command to be spawned as well as any
+/// necessary CLI args
+struct DecompressionCommand {
+    cmd: &'static str,
+    args: Vec<&'static str>
+}
+
+impl DecompressionCommand {
+    /// Create a new decompress command
+    fn new(cmd: &'static str, args: Vec<&'static str>) -> DecompressionCommand {
+        DecompressionCommand {
+            cmd, args
+        }
+    }
+}
+
+lazy_static! {
+    static ref DECOMPRESSION_COMMANDS: HashMap<&'static str, DecompressionCommand> = {
```

Comment:
why isn't this an enum?

better yet, can't this stuff be done without spawning subprocesses? i imagine there are rust libraries for most of these.

Reply:
The rationale for this implementation is explained in #225 and #539

There are still very few pure-Rust (de)compression libraries AFAIK